### ISSUE TITLE
PATCH Posting Twice to Console

### ISSUE THREAD

**Issue by danielleletarte** [ga-wdi-boston/game-project#523]
I have click counter on my function that updates the game board, but it seems to be skipping even numbers. When I review the console I see the clickCounter jumping from 1 to 3, 3 to 5 and so on. I set the clickCounter to 1 when I create a new game. Here is my code:

```
const populateSquare = function () {
  $(this).css('background-color', 'pink')
  const index = parseInt($(this).attr('id'))
  if (gameStore.game.clickCount % 2 === 1) {
    this.innerHTML = 'x'
    const value = 'x'
    const data = {
      'game': {
        'cell': {
          'index': index,
          'value': value
        },
        'over': false
      }
    }
    api.updateBoard(data)
      .then(ui.newMoveSuccess)
      .catch(ui.newMoveFailure)
  } else {
    this.innerHTML = 'o'
    const value = 'o'
    const data = {
      'game': {
        'cell': {
          'index': index,
          'value': value
        },
        'over': false
      }
    }
    api.updateBoard(data)
    .then(ui.newMoveSuccess)
    .catch(ui.newMoveFailure)
  }
  $(this).off('click')
}

const updateBoard = (data) => {
  return $.ajax({
    url: config.apiOrigin + '/games/' + gameStore.game.id,
    method: 'PATCH',
    headers: {
      Authorization: 'Token token=' + store.user.token
    },
    data
  })
}

const newMoveSuccess = (data) => {
  gameStore.game.clickCount++
  console.log(gameStore.game)
}

```

Here is the console results for a single game create and move:

```

2017-04-06 11:48:29.350 ui.js:18 Object {game: Object}game: Objectcells: Array(9)clickCount: 3id: 65over: falseplayer_o: nullplayer_x: Object__proto__: Object__proto__: Object
2017-04-06 11:48:33.808 jquery.js:9566 XHR finished loading: PATCH "http://localhost:4741/games/65".
send @ jquery.js:9566
ajax @ jquery.js:9173
updateBoard @ api.js:18
populateSquare @ events.js:30
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
2017-04-06 11:48:33.810 ui.js:29 Object {id: 65, cells: Array(9), over: false, player_x: Object, player_o: null…}cells: Array(9)0: ""1: ""2: ""3: ""4: ""5: ""6: ""7: ""8: ""length: 9__proto__: Array(0)clickCount: 3id: 65over: falseplayer_o: nullplayer_x: Object__proto__: Object
2017-04-06 11:48:33.812 jquery.js:9566 XHR finished loading: PATCH "http://localhost:4741/games/65".
send @ jquery.js:9566
ajax @ jquery.js:9173
updateBoard @ api.js:18
populateSquare @ events.js:30
dispatch @ jquery.js:5206
elemData.handle @ jquery.js:5014
2017-04-06 11:48:33.814 ui.js:29 

```

Might be easier to sit down and walk you through what I'm doing, but this is what I have right now.

**Comment by BenGitsCode** [comment]
Have you read the jquery documentation, specifically on the subtle and distinct differences between the `off` `on` and `one` methods? 

I suggest doing so on the jquery docs or within this issue queue. 

Let me know if that helps.

**Comment by danielleletarte** [comment]
I'm not sure how jquery is part of the issue I'm having? The click handler I have in place is working ok. If I comment it out, i'm still having the PATCH issue where it's posting twice.

**Comment by danielleletarte** [comment]
I'd like to sit down and review this with someone if I could. Been struggling with it for a while today. Thanks!

**Comment by benjimelito** [comment]
Coming over to take a look now

**Comment by payne-chris-r** [comment]
They're both sending different JSON. Do a search in your project for `api.updateBoard`?

**Comment by danielleletarte** [comment]
The issue had to do with how I grouped my click handlers for the whole page. I was calling the addHandlers function in the startNewGame function I created to reset them. Included in the Handlers function was the New Game click handler. This handler was removed and included in the handlers to be loaded when the page loads. The handlers for the game itself will load only when a new game is triggered. Fixed the issue, thanks Ben!
